// @tryghost/collections/build/CollectionsService.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CollectionsService = void 0;
class CollectionsService {
    repository;
    constructor(deps) {
        this.repository = deps.repository;
    }
    async save(data) {
        const collection = await this.repository.create(data);
        await this.repository.save(collection);
        return collection;
    }
    async edit(data) {
        const collection = await this.repository.getById(data.id);
        if (!collection) {
            return null;
        }
        Object.assign(collection, data);
        await this.repository.save(collection);
        return collection;
    }
    async getById(id) {
        return await this.repository.getById(id);
    }
    async getAll(options) {
        const collections = await this.repository.getAll(options);
        return {
            data: collections,
            meta: {
                pagination: {
                    page: 1,
                    pages: 1,
                    limit: collections.length,
                    total: collections.length,
                    prev: null,
                    next: null
                }
            }
        };
    }
    async destroy(id) {
        const collection = await this.getById(id);
        if (collection) {
            collection.deleted = true;
            await this.save(collection);
        }
    }
}
exports.CollectionsService = CollectionsService;
